rllib/algorithms/dt/dt.py

```diff
@@ -0,0 +1,361 @@
+
+        Args:
+            replay_buffer_config: Replay buffer config.
+                {
+                    "capacity": How many trajectories/episodes does the buffer hold.
+                }
+            embed_dim: Dimension of the embeddings in the GPT model.
+            num_layers: Number of attention layers in the GPT model.
+            num_heads: Number of attention heads in the GPT model. Must divide
+                embed_dim evenly.
+            embed_pdrop: Dropout probability of the embedding layer of the GPT model.
+            resid_pdrop: Dropout probability of the residual layer of the GPT model.
+            attn_pdrop: Dropout probability of the attention layer of the GPT model.
+            target_return: The target return-to-go for inference/evaluation.
+            grad_clip: If specified, clip the global norm of gradients by this amount.
+            lr_schedule: Learning rate schedule. In the format of
+                [[timestep, lr-value], [timestep, lr-value], ...]
+                Intermediary timesteps will be assigned to interpolated learning rate
+                values. A schedule should normally start from timestep 0.
+            **kwargs: Forward compatibility kwargs
+
+        Returns:
+            This updated DTConfig object.
+        """
+        super().training(**kwargs)
+        if replay_buffer_config is not None:
+            self.replay_buffer_config = replay_buffer_config
+        if embed_dim is not None:
+            self.embed_dim = embed_dim
+        if num_layers is not None:
+            self.num_layers = num_layers
+        if num_heads is not None:
+            self.num_heads = num_heads
+        if embed_pdrop is not None:
+            self.embed_pdrop = embed_pdrop
+        if resid_pdrop is not None:
+            self.resid_pdrop = resid_pdrop
+        if attn_pdrop is not None:
+            self.attn_pdrop = attn_pdrop
+        if target_return is not None:
+            self.target_return = target_return
+        if grad_clip is not None:
+            self.grad_clip = grad_clip
+        if lr_schedule is not None:
+            self.lr_schedule = lr_schedule
+
+        return self
+
+
+class DT(Algorithm):
+    """Implements Decision Transformer: https://arxiv.org/abs/2106.01345"""
+
+    # TODO: we have a circular dependency for get
+    #  default config. config -> Trainer -> config
+    #  defining Config class in the same file for now as a workaround.
+
+    @override(Algorithm)
+    def validate_config(self, config: AlgorithmConfigDict) -> None:
+        """Validates the Trainer's config dict.
+
+        Args:
+            config: The Trainer's config to check.
+
+        Raises:
+            ValueError: In case something is wrong with the config.
+        """
+        # Call super's validation method.
+        super().validate_config(config)
+
+        # target_return must be specified
+        assert (
+            self.config.get("target_return") is not None
```

Comment:
Target return should be part of the evaluation config dict I feel like. Not the training config.

Reply:
fixed